src/test/scala/com/gd/twitteranalytics/TweetsDataLoaderTest.scala

```diff
@@ -1,46 +1,44 @@
 package com.gd.twitteranalytics
 
-import java.io.File
 import java.util.{Date => JavaDate}
 import java.sql.Date
 
 import com.gd.twitteranalytics.util.AppConfigReader
 import com.holdenkarau.spark.testing.DataFrameSuiteBase
-import org.scalatest.FreeSpec
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.scalatest.{BeforeAndAfter, FreeSpec}
 
-class TweetsDataLoaderTest extends FreeSpec with DataFrameSuiteBase {
+class TweetsDataLoaderTest extends FreeSpec with DataFrameSuiteBase with BeforeAndAfter{
+
+    val payload = "This is the #Sample #text for #test purpose"
+    val event_date = new Date(new JavaDate("Wed Sep 26 10:37:20 PDT 2018").getTime)
+    val sampleDate = "2018-09-26"
+    val savePath = AppConfigReader.getAppConfigurables(0)+"/event_date="+s"${sampleDate}"
+
+  before{
+    val sqlCtx = sqlContext
+    import sqlCtx.implicits._
+
+    val tweetDataSet = Seq(Tweet(event_date,payload),Tweet(event_date,payload)).toDS
+    TweetsDataLoader.saveOutputToHdfs(tweetDataSet,savePath)
```

Comment:
a logic which is target for testing should be present in test case code-block, not extracted to `before` code-block for reuse.
Only operational stuff (e.g. data generation, initialization) should reside in `before`.
Otherwise, test cases become not straightforward. One should be able to reason about use cases of an API by looking into corresponding test cases.